### ISSUE TITLE
Not able to search from address bar: 400 Bad Request The plain HTTP request was sent to HTTPS port

### ISSUE THREAD

**Issue by githupper** [benbusby/whoogle-search#232]
Using personal domain certificate
Whoogle is hosted on Docker
HTTPS_ONLY set to 1
Using reverse proxy to redirect HTTPS to HTTP

Going to https://domain:port works fine in Firefox and Edge but if I type anything and search from address bar, it will throw 400 Bad Request The plain HTTP request was sent to HTTPS port.

Not sure what could be the issue. Please advise.

**Comment by shimuldn** [comment]
We do proxy redirect http > https.
Like this:

```
server {
        listen 80 default_server;
        listen [::]:80 default_server;
        server_name example.com;

       if ($host = example.com) {return 301 https://$host$request_uri;}
}
```

You check check

```
#ssl on 
OR
ssl off
```
`systemctl restart nginx`

**Comment by githupper** [comment]
Sorry I'm not that tech savvy so not sure how to handle your last comment. 

If it may help, I'm using Whoogle on my Synology docker and I'm using the build in reverse proxy on my Synology.

**Comment by shimuldn** [comment]
Need the reverse proxy config file of Nginx, caddy or whatever you use.
For Nginx check `cat /etc/nginx/sites-enabled/default` or `domain.conf `

**Comment by githupper** [comment]
Mine is the default reverse proxy functionality in Synology which expose limited configurable settings. 

Only thing that is configurable on there is point http to https or vice versa, IP/hostname, port number, Enable HSTS, Enable HTTP/2 and access control profile

I know it's not very helpful but that's all I can configure at this point.

**Comment by benbusby** [comment]
This might be something simpler than a config issue. Can you check to see what `<whoogle url>/opensearch.xml` is indicating as your root url? If it says `http://` rather than `https://` then it might just be a bug with completing the opensearch template on the back end.

**Comment by githupper** [comment]
Hi benbusby,

Is this what you mean by the root URL? If so, they are all in http and not https.

<Url type="text/html"  template="**http**://192.168.0.2:6000/search">
    <Param name="q" value="{searchTerms}"/>
  </Url>
  <Url type="application/x-suggestions+json"  template="**http**://192.168.0.2:6000/autocomplete">
      <Param name="q" value="{searchTerms}"/>
  </Url>
  <moz:SearchForm>**http**://192.168.0.2:6000/search</moz:SearchForm>
</OpenSearchDescription>

**Comment by benbusby** [comment]
Yes, that's what I needed. So just to clarify, searching with Whoogle using the webpage in your browser works, but not through the browser address bar? Also, does the `Root URL` value on the Whoogle home page's config menu specify `https://`? This might be fixable just by setting that to the correct value and then re-adding it as a search engine (you may need to clear the cache for the website beforehand, as the opensearch template is usually cached).

**Comment by githupper** [comment]
![image](https://user-images.githubusercontent.com/25658765/112585880-a7873900-8e35-11eb-94eb-aff418c8fbae.png)

I have tried removing cache and it did not work. Also GrapheneOS which supports automatically adding of new search engine, did not capture Whoogle as well.

Currently I have changed to a new IP and port and still did not get it to work with https currently.

So just to clarify, searching with Whoogle using the webpage in your browser works, but not through the browser address bar?  - that is correct.

**Comment by benbusby** [comment]
The URL in the screenshot still says `http://`, did you try updating that to `https://`? That value is what the opensearch template uses for generating the address bar search URL. Ultimately that value should get updated to match whatever URL you're wanting to search from. 

I have no way of testing GrapheneOS, so I can't help with that.

**Comment by githupper** [comment]
Yup. I changed that to https but reverted to http and also turn on HTTPS_ONLY as I wanted to be able to search from the address bar while I get advice from this Github thread.

I understood from GrapheneOS core developer that searching in OpenSearch supported search engines will automatically add it into the selectable search engines, much like FireFox but Whoogle did not appear inside when I tried it.
